src/main/java/com/google/firebase/auth/OidcProviderConfig.java

```diff
@@ -31,17 +34,31 @@
   @Key("clientId")
   private String clientId;
 
+  @Key("clientSecret")
+  private String clientSecret;
+
   @Key("issuer")
   private String issuer;
 
+  @Key("responseType")
+  private GenericJson responseType;
+
   public String getClientId() {
     return clientId;
   }
 
+  public String getClientSecret() {
+    return clientSecret;
+  }
+
   public String getIssuer() {
     return issuer;
   }
 
+  public GenericJson getResponseType() {
```

Comment:
This cannot really be part of the public API surface. Should use a more standard return type like Map or something. Or may be add two separate getters to mirror the setters you've added below:

```
boolean isCodeResponseType()
boolean isIdTokenResponseType()
```